### ISSUE TITLE
Cannot trigger debug config dropdown with keyboard

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode#21193]
Steps to reproduce:

1. Focus dropdown
2. enter/spacebar do nothing

![image](https://cloud.githubusercontent.com/assets/2193314/23227141/e7633c78-f8ec-11e6-9297-4db85ac22109.png)

**Comment by isidorn** [comment]
Enter/spacebar are designed to do nothing - it is exactly the same behavior like in all our dropdowns.
In order to get the drop down you need to use the up / down keys like in our other dropdowns.
I believe this is expected native behavior. Let me know if you disagree

**Comment by Tyriar** [comment]
We should probably change the other dropdown too then, this is not native on Linux and I don't think it is on Windows either. Basic test https://jsbin.com/senopunulo/edit?html,output

**Comment by isidorn** [comment]
The issue here is that our actionbar is eating up the space to trigger an action [here](https://github.com/Microsoft/vscode/blob/master/src/vs/base/browser/ui/actionbar/actionbar.ts#L437)

@bpasero why do we trigger actions on space - is this expected behavior?
@Tyriar in the example you provided on my mac enter still does not trigger the drop down, only space does.

**Comment by cleidigh** [comment]
the debugger drop down will activate (show drop-down items) by pressing alt-down
when the list item is active next to the debug start icon.
for Windows applications ( certainly things like MS office and other mainstream applications)
this is the default method for drop-down menus. in things like Excel up\down are not used
since they obviously change cell locations. can't comment on Lennox or Darwin.
but right now I think VSC does the right thing with the debug launch menu...

I do think the trigger actions on buttons like install for extensions could be altered
for a keyboard / voice user many keystrokes / commands have to be used to install an extension
primarily using tab stops which is not optimal.

**Comment by isidorn** [comment]
Due to the two comments above removing accessibility label and adding feature request.
More preciesly:
* users can interact with the dropdowns using up / down key
* users shuold be able to interact with the dropdown using space / enter, the issue here is the implemenation of the `SelectActionItem` which runs the action on space due to `actionBar` behavior which we should not change. We could solve this by decoupling more the `SelectActionItem` from the actual action it is tied so. So that on space the SelectActionItem does the native select behavior instead of running the action

[Code pointer](https://github.com/Microsoft/vscode/blob/master/src/vs/base/browser/ui/actionbar/actionbar.ts#L690)

**Comment by bpasero** [comment]
@cleidigh this would be addressed by https://github.com/Microsoft/vscode/pull/37533 at least for Windows and Linux, right?

**Comment by cleidigh** [comment]
@bpasero 
Yes, see keyboard triggers below:

			// Create and drop down select list on keyboard select
			switch (process.platform) {
				case 'darwin':
					if (event.keyCode === KeyCode.DownArrow || event.keyCode === KeyCode.UpArrow || event.keyCode === KeyCode.Space || event.keyCode === KeyCode.Enter) {
						showDropDown = true;
					}
					break;
				case 'win32':
				default:
					if (event.keyCode === KeyCode.DownArrow && event.altKey || event.keyCode === KeyCode.Space || event.keyCode === KeyCode.Enter) {
						showDropDown = true;
					}
					break;
			}

**Comment by bpasero** [comment]
Fyi pushed the change to use custom select box on Windows and Linux at least.

**Comment by cleidigh** [comment]
@Tyriar 
@isidorn 
@bpasero 
I am checking/clearing all issues associated with SelectBox, new or old.  I am unclear if I have addressed everything discussed here.  Issue title from @Tyriar  addresses keyboard triggers for the select.  I think I have whatever one wants for that, however, there are other standard triggers not mentioned above.  What is not clear to me is if it is desired that Space select items AFTER drop-down triggered.  Windows and Linux ONLY use Enter for item selection.  OS X can use either Enter or Space for item selection.

Should I add Space for item selection, SelectBox Custom on Windows/Linux ?

For Reference - All the following trigger the custom select box: (native standard)
- Enter
- Space
- Alt-Down
- Alt-Up - fix pending

**Comment by bpasero** [comment]
@cleidigh I think we should try to match the keyboard accessibility of the native control on any OS and leave it at that. I can see that on Windows at least pressing Enter or space expands a dropdown.

**Comment by bpasero** [comment]
@Tyriar seems to work fine now?

**Comment by Tyriar** [comment]
:+1:
